src/dynamodb/batch/duplicate-handling/filter.js

```diff
@@ -0,0 +1,102 @@
+function filterUniqueKeys(keys = []) {
+  if (!keys || !keys.length) return [];
+  if (keys.some((key) => typeof key !== 'object'))
+    throw new Error('All keys should must be objects.');
+
+  const stringifiedKeys = [];
+  const uniqueKeys = [];
+
+  keys.forEach((key) => {
+    if (
+      !stringifiedKeys.find((stringKey) => stringKey === JSON.stringify(key))
+    ) {
+      stringifiedKeys.push(JSON.stringify(key));
+      uniqueKeys.push(key);
+    }
+  });
+
+  return uniqueKeys;
+}
+
+const defaultDuplicateOptions = {
+  duplicateConfig: {
+    partitionKeyAttributeName: '',
+    sortKeyAttributeName: '',
+    timestampAttributeName: ''
+  }
+};
+
+function keyComparer(object1, object2, options) {
+  const pk = options.duplicateConfig.partitionKeyAttributeName;
+  const sk = options.duplicateConfig.sortKeyAttributeName;
+
+  if (sk) return object1[pk] === object2[pk] && object1[sk] === object2[sk];
+  return object1[pk] === object2[pk];
+}
+
+function objectAlreadySetAsUnique(uniqueObjects, object, options) {
+  return uniqueObjects.some((unique) => keyComparer(object, unique, options));
+}
+
+function getSortAttributeValues(a, b, sortAttribute) {
+  if (!sortAttribute.includes('.'))
+    return {
+      a: a[sortAttribute],
+      b: b[sortAttribute]
+    };
+
+  const firstSeparator = sortAttribute.indexOf('.');
+  const currentAttribute = sortAttribute.substr(0, firstSeparator);
+  const remainingAttributes = sortAttribute.substr(firstSeparator + 1);
+
+  return getSortAttributeValues(
+    a[currentAttribute],
+    b[currentAttribute],
+    remainingAttributes
+  );
+}
+
+function getLatestInstance(objects, currentObject, options) {
+  const sortAttribute = options.duplicateConfig.timestampAttributeName;
```

Comment:
Should this config attribute be renamed to something else? Could be something that is not timestamp, revision for instance?

Reply:
Revision indicates a number imo. Maybe `AttributeNameForLastUpdated`? Really tricky with a generic name so that's why I chose timestamp, but you're right it could be used with revision or other values that can be sorted desc.

Reply:
Ok, I will use AttributeNameForLastUpdated or something similar if I don't come up with something better

Reply:
`versionAttributeName` of course... A bit of the same issue as with `revision` but version feels more generic, I'll go with that